src/proxy/pool.rs

```diff
@@ -337,6 +336,7 @@ impl WorkloadHBONEPool {
     // Callers should then be safe to drop() the pool instance.
     pub fn new(
         cfg: Arc<crate::config::Config>,
+        original_source: bool,
```

Comment:
the idea is to convey that thru mutability, `config` isn't mutable, so it's by-definition fixed config/intent.

Reply:
Yeah that's why I think the config flag should just be two flags: https://github.com/istio/ztunnel/pull/1085#discussion_r1610589948

Reply:
as it is we have the advantage of not being able to specify an invalid state... what if they set require and disable to true?

Reply:
We can make it a 3 way enum - the invalid state is sort of an unescapable aspect of you telling us to do A but us maybe not being able to do it because of circumstances beyond our control.